dao/src/main/java/org/thingsboard/server/dao/sql/alarm/AlarmRepository.java

```diff
@@ -404,4 +392,25 @@ String updateAlarm(@Param("t_id") UUID tenantId, @Param("a_id") UUID alarmId, @P
     @Query(value = "DELETE FROM alarm_types AS at WHERE NOT EXISTS (SELECT 1 FROM alarm AS a WHERE a.tenant_id = at.tenant_id AND a.type = at.type) AND at.tenant_id = :tenantId AND at.type IN (:types)", nativeQuery = true)
     int deleteTypeIfNoAlarmsExist(@Param("tenantId") UUID tenantId, @Param("types") Set<String> types);
 
+    @Query(value = "SELECT a.id " +
```

Comment:
Please attach the result of `EXPLAIN ANALYZE` for this query. Should we include tenantId in the search?